src/main/java/io/obswebsocket/community/client/OBSCommunicator.java

```diff
@@ -227,9 +228,17 @@ public void onHello(Hello hello) {
                 hello.isAuthenticationRequired()
         ));
 
+        // If RPC version doesn't match, then the protocol isn't supported
+        if(hello.getRpcVersion() != RPC_VERSION) {
+            this.onError(session, new IllegalStateException(
```

Comment:
Aren't we supposed to throw that inside of `onIdentified` if the `negotiatedRpcVersion` is not the one we sent?

Reply:
Nope! 
We should be passing errors to onError so connections etc can be closed safely/consistently

Reply:
I think I was not being clear sorry.
Shouldn't we be doing this inside of `onIdentified` instead of here?
Because the OBSWebSocket server sends its latest RPC Version but is still compatible with previous version.
see https://github.com/Palakis/obs-websocket/blob/5cf2b50b633851f54e26b3a34c0ada9c14ed22c0/src/WebSocketServer.cpp#L278

Reply:
Ah! Okay, I see what you mean now, great point. 
It is even stated in the [connection steps](https://github.com/Palakis/obs-websocket/blob/master/docs/generated/protocol.md#connection-steps) the connection is dropped if rpc doesn't match in the Identify request. 
I'll update this and push the changes, thanks!